docs/UseCases.rst

```diff
@@ -0,0 +1,36 @@
+=====================
+Use Cases & Philosopy
+=====================
+
+.. warning:: Incorrect Information below
+
+    Until this document is reviewed and corrected, it is known to contain errors, especially w.r.t. philosopy.
+
+Philosopy
+=========
+
+Frost is intended to build upon the pytest framework to provide extra features to support "infrastructure auditing".
+
+The added features include:
+
+    - providing a standard way to inject meta data in the JSON test result file.
+    - providing a CLI wrapper (the ``frost`` program) to simplify execution of tests within the full supported frost environment.
+
+Note that the added features are all extensions to the normal pytest operations. Therefore, all tests can be executed directly from pytest. This can be highly useful for adhoc queries, and the development of new tests.
+
+Target Domain
+-------------
+
+Frost is intended to simplify the process of auditing existing system configurations, when the configuration data is accessible via an API. Frost assumes that the test results (in JSON format) will be acted upon by a subsequent non-Frost process.
+
+Since, in general, the number of configuration items to check is not statically known when the test is writen or executed, some lesser used features of pytest are needed. In particular, the test collection phase often requires API calls to the service to discover the complete tree (or network) of configuration items to be checked. While uncommon in internet examples, dynamic test discovery is fully supported by pytest.
+
+In practice, many of the same calls needed to determine the list of items may need to be repeated to collect the additional information needed by one or more tests. This leads to a tension between collecting all potentially needed data in the first (discovery) pass, or making many duplicate calls to the service as each test collects the additional information it requires.
+
+Use Cases
```

Comment:
I'm confused by what this sections is about. Is this "Use Cases for Frost"?

Reply:
Yes.